expression/builtin_time.go

```diff
@@ -1965,7 +1966,28 @@ type builtinTimeToSecSig struct {
 
 // See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_time-to-sec
 func (b *builtinTimeToSecSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("TIME_TO_SEC")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+
+	t, err := args[0].ToString()
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+
+	curTime := []int64{}
+	for _, v := range strings.Split(t, ":") {
+		int64V, err := strconv.ParseInt(v, 10, 64)
+		if err != nil {
+			return d, errors.Trace(err)
+		}
+		curTime = append(curTime, int64V)
+	}
+
+	d.SetInt64(curTime[0]*3600 + curTime[1]*60 + curTime[2])
```

Comment:
I changed to use `convertToDuration` to do this job, so I don't need to do that chenk any more.